model-optimizer/extensions/front/OneHotDepthNormalizer.py

```diff
@@ -0,0 +1,43 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementPattern
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph
+from mo.ops.reshape import Reshape
+
+
+class OneHotDepthNormalizer(FrontReplacementPattern):
+    """
+    Transformation performs safe replacement of the 0D constants with 1D for a specific operations. This transformation
+    allows to avoid problems with the fact that not all layers correctly handle 0D tensors during the constant folding.
+    """
+    enabled = True
+    force_clean_up = True
```

Comment:
`force_clean_up` is not needed in this transformation. Moreover, it is dangerous on a front phase.